cpp/src/io/orc/writer_impl.cu

```diff
@@ -1980,37 +2017,34 @@ void writer::impl::write(table_view const& table)
       task.wait();
     }
 
-    if (not column_stats.empty()) {
-      // File-level statistics
-      // NOTE: Excluded from chunked write mode to avoid the need for merging stats across calls
-      if (single_write_mode) {
-        // First entry contains total number of rows
-        buffer_.resize(0);
-        pbw_.putb(1 * 8 + PB_TYPE_VARINT);
-        pbw_.put_uint(num_rows);
-        ff.statistics.reserve(1 + orc_table.num_columns());
-        ff.statistics.emplace_back(std::move(buffer_));
-        // Add file stats, stored after stripe stats in `column_stats`
-        ff.statistics.insert(
-          ff.statistics.end(),
-          std::make_move_iterator(column_stats.begin()) + stripes.size() * orc_table.num_columns(),
-          std::make_move_iterator(column_stats.end()));
-      }
-      // Stripe-level statistics
+    // File-level statistics
+    // NOTE: Excluded from chunked write mode to avoid the need for merging stats across calls
+    if (single_write_mode and not statistics.file_level.empty()) {
+      // First entry contains total number of rows
+      buffer_.resize(0);
+      pbw_.put_byte(1 * 8 + PB_TYPE_VARINT);
```

Comment:
more magic numbers here. What does the `1 * 8` represent? Is this a bitmask?

Reply:
That's how the field numbers in protobuf are encoded, with last three bits reserved the field type. 
Centralized the magic into a single function (improvement over the pre-PR code:)), let me know what you think.

Reply:
Yes, I was reluctant to suggest since it was old code, but it seemed a small enough change and enough of a win. I like how this turned out. Thank you for doing this extra work.